# 🧹 Limpeza e Preparação de Dados — Olist E-commerce

Dataset público da Olist com dados reais de e-commerce brasileiro.
Este notebook realiza a limpeza e preparação de todas as tabelas antes das análises.

In [2]:
import pandas as pd


pedidos    = pd.read_csv("../dados/olist_orders_dataset.csv")
clientes   = pd.read_csv("../dados/olist_customers_dataset.csv")
itens      = pd.read_csv("../dados/olist_order_items_dataset.csv")
pagamentos = pd.read_csv("../dados/olist_order_payments_dataset.csv")
produtos   = pd.read_csv("../dados/olist_products_dataset.csv")

print("Pedidos:", pedidos.shape)
print("Clientes:", clientes.shape)
print("Itens:", itens.shape)
print("Pagamentos:", pagamentos.shape)
print("Produtos:", produtos.shape)

Pedidos: (99441, 8)
Clientes: (99441, 5)
Itens: (112650, 7)
Pagamentos: (103886, 5)
Produtos: (32951, 9)


## 📋 Tabela Principal — Pedidos (olist_orders)

Tabela central do modelo. Contém todos os pedidos realizados na plataforma,
com status, datas de compra, aprovação e entrega.

In [8]:
pedidos.shape

(99441, 8)

### Verificando tipos de dados


In [8]:
pedidos.dtypes

order_id                         str
customer_id                      str
order_status                     str
order_purchase_timestamp         str
order_approved_at                str
order_delivered_carrier_date     str
order_delivered_customer_date    str
order_estimated_delivery_date    str
dtype: object

### 🗓️ Conversão de Datas

In [3]:
coluna_datas = [
'order_purchase_timestamp','order_approved_at','order_delivered_carrier_date',
'order_delivered_customer_date','order_estimated_delivery_date'
]

for coluna in coluna_datas:
    pedidos[coluna] = pd.to_datetime(pedidos[coluna])

pedidos.dtypes

order_id                                    str
customer_id                                 str
order_status                                str
order_purchase_timestamp         datetime64[us]
order_approved_at                datetime64[us]
order_delivered_carrier_date     datetime64[us]
order_delivered_customer_date    datetime64[us]
order_estimated_delivery_date    datetime64[us]
dtype: object

### 🔍 Verificação de Valores Nulos

In [14]:
pedidos.isnull().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

### 🛠️ Tratamento de Valores Nulos

- `order_approved_at` — 160 nulos — pedidos não aprovados
- `order_delivered_carrier_date` — 1.783 nulos — pedidos não coletados pela transportadora  
- `order_delivered_customer_date` — 2.965 nulos — pedidos não entregues ao cliente

Os nulos são esperados — representam pedidos em andamento ou cancelados.
Serão mantidos pois filtraremos por `order_status = 'delivered'` nas análises.

In [15]:
pedidos['order_status'].value_counts()

order_status
delivered      96478
shipped         1107
canceled         625
unavailable      609
invoiced         314
processing       301
created            5
approved           2
Name: count, dtype: int64

In [16]:
pedidos.shape

(99441, 8)

### ✅ Resultado Final — Pedidos

- **96.478 pedidos entregues** — base principal das análises
- Nulos nas colunas de data são esperados — pedidos não entregues
- Nenhum registro removido — todos os status serão mantidos para análise completa

## 📋 Tabela de Itens — (olist_order_items)

Contém os itens de cada pedido com produto, preço e frete.

In [11]:
itens.head(1)

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.9,13.29


### Verificando tipos de dados

In [12]:
itens.dtypes

order_id                   str
order_item_id            int64
product_id                 str
seller_id                  str
shipping_limit_date        str
price                  float64
freight_value          float64
dtype: object

### 🗓️ Conversão de Datas

In [4]:
itens['shipping_limit_date'] = pd.to_datetime(itens['shipping_limit_date'])
itens.dtypes

order_id                          str
order_item_id                   int64
product_id                        str
seller_id                         str
shipping_limit_date    datetime64[us]
price                         float64
freight_value                 float64
dtype: object

### 🔍 Verificação de Valores Nulos

In [17]:
itens.isnull().sum()

order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

### ✅ Resultado Final — Itens

- Nenhum valor nulo encontrado
- Data `shipping_limit_date` convertida para datetime
- Tabela pronta para análise

## 📋 Tabela de Clientes — (olist_customers)

Contém informações de localização dos clientes por cidade e estado.

### 🔍 Tipos de Dados

In [18]:
clientes.head(1)

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP


In [19]:
clientes.dtypes

customer_id                   str
customer_unique_id            str
customer_zip_code_prefix    int64
customer_city                 str
customer_state                str
dtype: object

### 🔍 Valores Nulos

In [20]:
clientes.isnull().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

### ✅ Resultado Final — Clientes

- Nenhum valor nulo encontrado
- Nenhuma conversão de tipo necessária
- Tabela pronta para análise

## 📋 Tabela de Pagamentos — (olist_order_payments)

Contém informações de pagamento de cada pedido, incluindo tipo e valor.

### 🔍 Tipos de Dados

In [21]:
pagamentos.dtypes

order_id                    str
payment_sequential        int64
payment_type                str
payment_installments      int64
payment_value           float64
dtype: object

In [22]:
pagamentos.head(1)

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33


### 🔍 Valores Nulos

In [23]:
pagamentos.isnull().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

### ✅ Resultado Final — Pagamentos

- Nenhum valor nulo encontrado
- Nenhuma conversão de tipo necessária
- Tabela pronta para análise

## 📋 Tabela de Produtos — (olist_products)

Contém informações cadastrais dos produtos como categoria, dimensões e peso.

### 🔍 Tipos de Dados

In [1]:
produtos.dtypes

NameError: name 'produtos' is not defined

In [ ]:
produtos.head(1)

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0


### 🔍 Valores Nulos

In [ ]:
produtos.isnull().sum()


product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

### 🛠️ Tratamento de Valores Nulos

- 610 nulos em `product_category_name` — produtos sem categoria cadastrada
- 610 nulos em atributos descritivos do produto
- 2 nulos em peso e dimensões físicas — registros incompletos

**Estratégia:**

- `product_category_name` → preencher com `'Sem Categoria'`
- Atributos numéricos → preencher com a mediana da respectiva coluna

A mediana foi escolhida para os atributos numéricos por ser menos influenciada por valores extremos (outliers), podendo representar melhor o valor central da distribuição.

In [ ]:
# Preenchendo categoria nula
produtos['product_category_name'] = produtos['product_category_name'].fillna('Sem Categoria')

# Preenchendo dimensões físicas com mediana

colunas_medianas = [
     'product_name_lenght',
    'product_description_lenght', 
    'product_photos_qty',
    'product_weight_g',
    'product_length_cm',
    'product_height_cm',
    'product_width_cm'
]

for coluna in colunas_medianas:
    produtos[coluna] = produtos[coluna].fillna(produtos[coluna].median())

#COnfirmando
produtos.isnull().sum()

product_id                    0
product_category_name         0
product_name_lenght           0
product_description_lenght    0
product_photos_qty            0
product_weight_g              0
product_length_cm             0
product_height_cm             0
product_width_cm              0
dtype: int64

### converte product_photos_qty para inteiro

In [ ]:
produtos['product_photos_qty'] = produtos['product_photos_qty'].astype(int)

produtos.dtypes

product_id                        str
product_category_name             str
product_name_lenght           float64
product_description_lenght    float64
product_photos_qty              int64
product_weight_g              float64
product_length_cm             float64
product_height_cm             float64
product_width_cm              float64
dtype: object


## 💾 Exportando Dados Limpos

In [7]:
pedidos.to_csv("../dados/pedidos_limpo.csv", index=False)
clientes.to_csv("../dados/clientes_limpo.csv", index=False)
itens.to_csv("../dados/itens_limpo.csv", index=False)
pagamentos.to_csv("../dados/pagamentos_limpo.csv", index=False)
produtos.to_csv("../dados/produtos_limpo.csv", index=False)

print("✅ Dados exportados com sucesso!")

✅ Dados exportados com sucesso!
